### DISCUSSION TITLE
Google Play 'resubscribe' expired subscription

### DISCUSSION THREAD

**Discussion by iammart** [imdhemy/laravel-in-app-purchases#38] [Q&A]
Hi,

How are people handling the event `SUBSCRIPTION_PURCHASED` when a user hits "resubscribe" from the google play store for a subscription that was cancelled and has now expired (but still in the window of renewal/resubscribe)?

This event fails to contain any information regarding any previous/original subscription, which makes it impossible for the configured listener to query the database for the user to whom to attach the new subscription too?

**Comment by imdhemy** [comment]
Hi @iammart,
Once received a new subscription:
- Verify (Get request). A success-response means it's valid.
- Acknowledge.

Is the `linkedPurchaseToken` missing even after verifying and acknowledging the new subscription?

**Reply by iammart** [reply]
Hey @imdhemy 

For now I am just concentrating on Google Play as I have an Android device I can test with...

`linkedPurchaseToken` is missing from the original request after resubscribe via Google Play Store

After having acknowledged the new subscription, executing a `get` on the `Subscription` to return a `SubscriptionPurchase` still fails to return a `linkedPurchaseToken`.

The only time I have been able log a value for `linkedPurchaseToken` is from buying a new subscription via the app when an original subscription was cancelled but not expired.

**Reply by imdhemy** [reply]
Hi @iammart,
>After having acknowledged the new subscription, executing a get on the Subscription to return a SubscriptionPurchase still fails to return a linkedPurchaseToken.

I mean two separate requests, the first is `acknowledge` and the latter is `get`.

```php
Subscription::googlePlay()->id($itemId)->token($token)->acknowledge();
$subscriptionReceipt = Subscription::googlePlay()->id($itemId)->token($token)->get();
``` 
My question is, is the `linkedPurchaseToken` missing even after executing both requests?

**Reply by iammart** [reply]
Sure, I understood that - sorry if that didn't come across correctly...

This morning I optimistically logged exactly this and unfortunately the `linkedPurchaseToken` is still null on the `SubscriptionPurchase` object that is returned from the `get` you describe above.

**Reply by iammart** [reply]
I have found open issues regarding this.

https://issuetracker.google.com/issues/165824685
https://issuetracker.google.com/issues/159667773

But finally ended up on the following issue tracker.

https://issuetracker.google.com/issues/176457322

Long story short, it appears to be functionality that does not exist or will exist any time soon. It's only ever present when the subscription is renewed in app. Therefore this needs delegating to the app as per docs here.

https://developer.android.com/google/play/billing/integrate#ooap
